src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -195,7 +230,12 @@ def tdr(self):
     @classmethod
     @cache_per_thread
     def _tdr(cls):
-        return TDRClient()
+        return TDRClient.with_service_account_credentials()
+
+    @classmethod
+    @cache_per_thread
+    def __source_cache(cls):
```

Comment:
Typically, services are used by controllers. The plugins don't use services. Think about why that is desirable.

Return type hint missing.

Reply:
Motivate the caching per thread. Service typically are NOT cached per thread.

Reply:
I don't remember what my motivation was.